cassandra/src/test/java/brave/cassandra/ITTracing.java

```diff
@@ -13,40 +13,56 @@
  */
 package brave.cassandra;
 
-import brave.cassandra.driver.CassandraClientTracing;
-import brave.cassandra.driver.TracingSession;
-import brave.propagation.CurrentTraceContext.Scope;
+import brave.propagation.TraceContext;
 import brave.test.ITRemote;
-import cassandra.CassandraRule;
+import cassandra.CassandraContainer;
+import cassandra.ForwardHttpSpansToHandler;
 import com.datastax.driver.core.BoundStatement;
 import com.datastax.driver.core.Cluster;
 import com.datastax.driver.core.Session;
-import com.datastax.driver.core.Statement;
+import java.nio.ByteBuffer;
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.function.Function;
-import org.junit.ClassRule;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.Timeout;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.images.RemoteDockerImage;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.MountableFile;
+import zipkin2.reporter.Reporter;
+import zipkin2.reporter.brave.ZipkinSpanHandler;
+import zipkin2.reporter.urlconnection.URLConnectionSender;
 
-import static brave.Span.Kind.CLIENT;
 import static brave.Span.Kind.SERVER;
-import static brave.propagation.SamplingFlags.NOT_SAMPLED;
+import static brave.propagation.B3SingleFormat.writeB3SingleFormatAsBytes;
+import static brave.propagation.SamplingFlags.SAMPLED;
+import static java.util.Collections.singletonMap;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class ITTracing extends ITRemote {
-  static {
-    System.setProperty("cassandra.custom_tracing_class", Tracing.class.getName());
-  }
+  /** Builds a Cassandra image that adds {@link Tracing} and its dependencies in the classpath. */
+  static RemoteDockerImage imageWithBraveTracer = new RemoteDockerImage(imageWithBraveTracer());
+
+  @Rule public ForwardHttpSpansToHandler zipkin = new ForwardHttpSpansToHandler(testSpanHandler);
+  @Rule public CassandraContainer cassandra = new CassandraContainer(imageWithBraveTracer)
```

Comment:
this takes so long to start cassandra for each test, I'm probably going to indirect access to testSpanHandler to avoid this.